### DISCUSSION TITLE
What is the use case for tag or tag-toggle command

### DISCUSSION THREAD

**Discussion by khanakia** [gokcehan/lf#1610] [Q&A] [answered]
I could not understand the use case for a tag or tag-toggle command.

**Comment by DusanLesan** [comment, marked as answer]
My use case is similar to having bookmarks. For example, when I'm watching a TV show with all episodes in one directory, I always tag the last watched episode so I know where I left off. Similarly, when reviewing several log files with similar names in a single directory, I tag the one I reviewed last. I use tags as a visual indicator to know where I need to continue some action between similarly named files.

Marks, on the other hand, I use as temporary jump points to files I will be using frequently for a period of time.

**Reply by thetaoinwork** [reply]
is it possible to use other characters for different tags rather than always using the default which is asterisk?

**Reply by DusanLesan** [reply]
Yes. You just need to provide that character as a argument for `tag` command. 

I wrote this now:
```
map T %{{
	echo "Single tag char: "
	read ans
	lf -remote "send $id tag $ans"
}}
```
When pressing on `T`, it prompts you for a character, and then applies it as a tag

**Reply by thetaoinwork** [reply]
Thank you so much, don't mean to take up so much of your time but do you think it would be easy to export an env variable like $fs that would contain all the files marked with a specific tag in the current working directory?

**Reply by DusanLesan** [reply]
Not difficult apparently.
```
map T %{{
	echo "Search tag char: "
	read query

	while IFS= read -r line; do
		path=$(echo $line | cut -d ':' -f 1)
		tag=$(echo $line | cut -d ':' -f 2)

		if [ "$tag" == "$query" ] && [ "$(dirname $path)" == "$PWD" ]; then
			lf -remote "send $id toggle \"$path\""
		fi
	done < "$HOME/.local/share/lf/tags"
}}
```
This searches for a specified tags in current dir and  selects them.
Tags are persisted in `~/.local/share/lf/tags`. So you are able to parse it. Instead of selecting like I did with `lf -remote "send $id toggle \"$path\""`, you can append values to a variable and export after while loop

**Reply by DusanLesan** [reply]
Instead of
```
path=$(echo $line | cut -d ':' -f 1)
tag=$(echo $line | cut -d ':' -f 2)
```
it is possible to do
```
IFS=':' read -r path tag <<< "$line
```
